### ISSUE TITLE
Incorrect severity level for a rule called from other rule

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#2139]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

If rule was executed from another rule, it has incorrect severity level `ignore`. stylelint-order's `declaration-block-property-groups-structure` [calling](https://github.com/hudochenkov/stylelint-order/blob/master/rules/declaration-block-property-groups-structure/index.js#L32) `declaration-block-properties-order`. `declaration-block-property-groups-structure` has default `error` severity level, but `declaration-block-properties-order` has `ignore` level. I've [added temporary fix](https://github.com/hudochenkov/stylelint-order/blob/master/rules/declaration-block-property-groups-structure/index.js#L34-L37) in my plugin to set severity level. I wanted to use `options` object for that, but didn't figured out how to get correct severity level. There is an `if` because when I rune tests there is no `result.stylelint.ruleSeverities`.

> What CSS is needed to reproduce this issue?

```css
a {
    width: 0;

    position: relative;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "order/declaration-block-property-groups-structure": [
            {
                "emptyLineBefore": "always",
                "order": "strict",
                "properties": [
                    "position",
                    "top"
                ]
            },
            {
                "emptyLineBefore": "always",
                "order": "strict",
                "properties": [
                    "width"
                ]
            }
        ]
    }
}
```

> Which version of stylelint are you using?

`7.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI, Node API.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

Called rule has same severity level as rule which called it.

> What actually happened (e.g. what warnings or errors you are getting)?

```js
{
    line: 5,
    column: 4,
    rule: 'declaration-block-properties-order',
    severity: 'ignore',
    text: 'Expected "position" to come before "width" (declaration-block-properties-order)'
}
```

**Comment by jeddy3** [comment]
@hudochenkov Thank you for the report. This looks like it'll require some investigation and any help would be appreciated.

**Comment by davidtheclark** [comment]
Uh oh ... maybe the ability to use other rules is not quite mature. I have an idea ...

**Comment by hudochenkov** [comment]
I believe this cause the problem https://github.com/stylelint/stylelint/blob/master/lib/utils/report.js#L59 But I don't know how to apply `defaultSeverity` like this https://github.com/stylelint/stylelint/blob/master/lib/lintSource.js#L106, because there is no `config` in report.js.

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/2173

**Comment by davidtheclark** [comment]
@hudochenkov In this case I suggest that you extract the shared logic into a function that you can invoke in both places. We do that with some rules in stylelint, too.
